crates/pink-libs/subrpc/src/lib.rs

```diff
@@ -233,19 +221,23 @@ pub fn create_transaction<T: Encode>(
     pallet_id: u8,
     call_id: u8,
     data: T,
+    extra: ExtraParam,
 ) -> core::result::Result<Vec<u8>, Error> {
     let version = get_ss58addr_version(chain)?;
     let public_key: [u8; 32] = signing::get_public_key(signer, SigType::Sr25519)
         .try_into()
         .unwrap();
     let addr = public_key.to_ss58check_with_version(version.prefix());
-    let nonce = get_next_nonce(rpc_node, &addr)?.next_nonce;
     let runtime_version = get_runtime_version(rpc_node)?;
     let genesis_hash: [u8; 32] = get_genesis_hash(rpc_node)?.0;
     let spec_version = runtime_version.spec_version;
     let transaction_version = runtime_version.transaction_version;
-    let era = Era::Immortal;
-    let tip: u128 = 0;
+    let era = extra.era.unwrap_or(Era::default());
```

Comment:
@kvinwang Do you think we can reuse the logic we implemented in subxt here?

Note that IMMORTAL will be very dangerous, since it never expire.

Reply:
Also `Immortal ` will force having genesis hash embedded is expensive because we fetch it from the blockchain every time we construct transaction data. Is there any link I can check?

Reply:
Here is how the pherry constructing the Era: https://github.com/Phala-Network/phala-blockchain/blob/674d1b7d1837b9e81bae56fcd4ccc56cb292de34/standalone/pherry/src/lib.rs#L1387-L1390

Reply:
Get it. Since now the user can customize `Era`, I prefer to let the application layer set the value they want

Reply:
>  I prefer to let the application layer set the value they want

Agree, a value of Mortal would rely on the current block number, which might be better to be fetched by the user side.